lib/battle/BattleHex.cpp

```diff
@@ -184,19 +184,65 @@ void BattleHex::checkAndPush(BattleHex tile, std::vector<BattleHex> & ret)
 		ret.push_back(tile);
 }
 
+uint8_t BattleHex::getDistance(const BattleHex & distanceTo) const
+{
+	return BattleHex::getDistance(*this, distanceTo);
+}
+
+BattleHex BattleHex::getWallHex(int y)
+{
+	static constexpr std::array<int16_t, 11> lineToHex = {12, 29, 45, 62, 78, 96, 112, 130, 147, 165, 182};
+
+	return BattleHex(lineToHex[y]);
+}
+
+BattleHex BattleHex::getWallHex() const
+{
+	return BattleHex::getWallHex(getY());
+}
+
+std::vector<BattleHex> BattleHex::getShortestPath(const BattleHex & pathTo) const

```

Comment:
@IvanSavenko then it will be static, because now it requires this.

Reply:
you know, you simply can NOT write static specifier.

Problem solved!

Reply:
Why? Why I need to move this, add new parameter and not to make method static?

Reply:
I am tired of this.
NO
STATIC
UNLESS
REQUIRED

No need to expose internal details (whether or not method iuses state) to public interface of the class.

Exceptions are singleton / factory method and such.

Reply:
If battle hex is overloaded, maybe create some namespace, like BattleHexHelpers, and move such methods there? About static - if you such strongly dislike static, maybe add this statement to project code style?

Reply:
1) It is not related to battle hex in any way
2) Eventually we may want to allow some other way to define walls position, one that won't be stateless, e.g. load from json. No need to make future changes more complex than necessary

So please, non-static method in BattleInfoCallback or other related class

> maybe add this statement to project code style?

I could, but half of that guide needs review / update, and half of our actual code style is not included in there.

Reply:
About getWallHex - agree. But why getShortestPath is unrelated to BattleHex?

Reply:
Because it is battle**Hex**. Not Battle**Field**.

It already has some battlefield-related functionality, but at the very least that functionality is limited to direct neighbors, and not full-fledged path builder.

Reply:
This method is not needed now